shared/domains/chelonia/chelonia.js

```diff
@@ -763,14 +771,11 @@ export default (sbp('sbp/selectors/register', {
   },
   // 'chelonia/contract' - selectors related to injecting remote data and monitoring contracts
   // TODO: add an optional parameter to "retain" the contract (see #828)
-  'chelonia/contract/sync': function (contractIDs: string | string[], params?: { force?: boolean, deferredRemove?: boolean }): Promise<*> {
+  'chelonia/contract/sync': function (contractIDs: string | string[], params?: { force?: boolean, resync?: boolean }): Promise<*> {
```

Comment:
If a contract hasn't been sync'd before, the retain count should probably be set to `1`.

Reply:
Based on discussion, the idea would be to add an extra parameter called `_foreign` (with an underscore to indicate it's a internal parameter, not to be used by developers). If it is set, then the persistent retain count is set to `0`, otherwise, if we are subscribing to the contract for the first time, it's set to `1`

Reply:
Based on discussion, the idea would be to add an extra parameter called something like `_ignoreRetainCount` (with an underscore to indicate it's a internal parameter, not to be used by developers).

- If it is not set, then the persistent retain count is set to `1` if we are subscribing to the contract for the first time.
- If it is set, the retain count is left unmodified to avoid screwing up foreign contracts

Reply:
Update: I'm admitting defeat here, and am reverting my position. My original attempt was to try to get rid of `try` in `release`. But I couldn't do that because of arguments @corrideat made related to recursive foreign contracts.

So then I thought maybe at least we can get `sync` to behave more like `alloc] init]` in ObjC, but looking at the changes requested here, this would be strictly a complication of the code, not a simplification.

So therefore, we're going to leave the approach as-is, and state in the documentation that `contract/sync` should never be called without first calling `retain`, and that `retain` should be used in place of `contract/sync` when syncing a contract for the first time.